src/components/Product/ProductList.js

```diff
@@ -0,0 +1,100 @@
+import {useState, useEffect} from "react";
+import {ProductListForm} from "./ProductListForm";
+import {ProductData} from "./ProductData";
+import axios from "axios";
+import {useCart} from "../../context/cart-context";
+
+export function ProductList() {
+  const [sliderValue, setSliderValue] = useState(1200);
+  const [error, setError] = useState(null);
+  const {dispatch} = useCart();
+  useEffect(() => {
+    const anonFun = async () => {
+      try {
+        const getProducts = await axios.get(
+          "https://ecom.amansethi00.repl.co/products"
+        );
+        console.log(getProducts);
+        if (getProducts.data.success) {
+          dispatch({type: "SET_PRODUCTS", payload: getProducts.data});
+        } else {
+          setError(getProducts.data.message);
+        }
+      } catch (error) {
+        setError(error);
+      }
+    };
+    anonFun();
+  }, []);
+
+  useEffect(() => {
+    const anonFun = async () => {
```

Comment:
Extract the API call into a function `fetchCart()`. Do the same for the wishlist API call and call both the functions in one `useEffect` using an **IIFE**

```suggestion
( 
    async () => {
        await fetchCart() 
        await fetchWishlist()
    }
)()
```